lib/min_heap.rb

```diff
@@ -58,14 +65,35 @@ def empty?
   # Time complexity: ?
   # Space complexity: ?
   def heap_up(index)
-    
+    return if !index || !@store[index]
+
+    parent = (index - 1) / 2
+
+    if @store[index].key < @store[parent].key
+      swap(index, parent)
+      return if parent == 0
+      heap_up(parent)
+    end
   end
 
   # This helper method takes an index and 
   #  moves it up the heap if it's smaller
   #  than it's parent node.
   def heap_down(index)
```

Comment:
Nice, very clever way to handle checking if the right and left children are beyond the array.